src/Expr.cc

```diff
@@ -3789,12 +3788,12 @@ RecordCoerceExpr::RecordCoerceExpr(ExprPtr arg_op, RecordTypePtr r)
 		RecordType* t_r = type->AsRecordType();
 		RecordType* sub_r = op->GetType()->AsRecordType();
 
-		map_size = t_r->NumFields();
-		map = new int[map_size];
+		int map_size = t_r->NumFields();
+		map.reserve(map_size);
 
 		int i;
 		for ( i = 0; i < map_size; ++i )
-			map[i] = -1;	// -1 = field is not mapped
+			map.emplace_back(-1);	// -1 = field is not mapped
```

Comment:
This block could be `std::fill_n(map.begin(), map_size, -1)`.

Reply:
Nifty, didn't know about that.

Reply:
Actually, turns out that to use this requires a preceding `resize()`, and that allows this particular operation directly.